src/puppetlabs/comidi.clj

```diff
@@ -167,11 +167,15 @@
     (update-in route-info [:path] conj "!")
 
     (sequential? pattern)
-    (if-let [next (first pattern)]
-      (update-route-info*
-        (update-in route-info [:path] conj next)
-        (rest pattern))
-      route-info)
+    (loop [info route-info
+           remaining pattern]
+      (let [head (first remaining)
```

Comment:
In order to avoid this let block, you can destructure this within the loop

instead of `remaining pattern` you can write `[head & tail] pattern` inside the loop form, and they should contain the same data

Reply:
Oh cool, I'll give that a try.

Reply:
Oh oops there's small difference: if you use that destructuring `tail` will be nil when there's only one thing left in `pattern`. but `(empty? nil)` is still true so it'll work, just might be a bit surprising, so you could change the check to be `(nil? tail)`

Reply:
Thanks, updated.